src/Factory.cc

```diff
@@ -227,33 +236,42 @@ void Factory::Register(const std::string &_msgType,
 std::unique_ptr<google::protobuf::Message> Factory::New(
     const std::string &_msgType)
 {
+  auto msgType = _msgType;
+  if (msgType.find("ignition") == 0)
+  {
+    msgType.replace(0, 8, "gz");
+    std::cerr << "Trying to create deprecated message type ["
+              << _msgType << "]. Use [" << msgType << "] instead."
+              << std::endl;
+  }
+
   std::unique_ptr<google::protobuf::Message> msg;
 
   std::string type;
   // Convert "gz.msgs." to "gz_msgs.".
-  if (_msgType.find("gz.msgs.") == 0)
+  if (msgType.find("gz.msgs.") == 0)
   {
-    type = "gz_msgs." + _msgType.substr(8);
+    type = "gz_msgs." + msgType.substr(8);
   }
   // Convert ".gz.msgs." to "gz_msgs.".
-  else if (_msgType.find(".gz.msgs.") == 0)
+  else if (msgType.find(".gz.msgs.") == 0)
   {
-    type = "gz_msgs." + _msgType.substr(9);
+    type = "gz_msgs." + msgType.substr(9);
   }
   else
   {
     // Fix typenames that are missing "gz_msgs." at the beginning.
-    if (_msgType.find("gz_msgs.") != 0)
+    if (msgType.find("gz_msgs.") != 0)
       type = "gz_msgs.";
-    type += _msgType;
+    type += msgType;
   }
 
   // Create a new message if a FactoryFn has been assigned to the message type
   if (msgMap->find(type) != msgMap->end())
     return ((*msgMap)[type]) ();
 
   // Check if we have the message descriptor.
-  msg = dynamicFactory.New(_msgType);
+  msg = dynamicFactory.New(msgType);
```

Comment:
If we're returning `gz`, should be the deprecation warning be "**Using** [gz.XXX]" instead of "Use [gz.XXX]"?